streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -501,14 +501,14 @@ private void runLoop() {
                     mainConsumer.enforceRebalance();
                 }
             } catch (final TaskCorruptedException e) {
-                log.warn("Detected the states of tasks {} are corrupted. " +
-                    "Will close the task as dirty and re-create and bootstrap from scratch.", e.corruptedTaskWithChangelogs());
+                log.warn("Detected the states of tasks " + e.corruptedTaskWithChangelogs() + " are corrupted. " +
+                             "Will close the task as dirty and re-create and bootstrap from scratch.", e);
```

Comment:
Rearranged this so we can include the exception itself. The stacktrace is useful for tracking down the reason for the exception.